services/harmony/test/models/data-operation.ts

```diff
@@ -378,6 +438,10 @@ describe('DataOperation', () => {
         expect(operation.model.sources[0].variables[0].fullPath).to.equal('the/nested/name');
       });
 
+      it('sets the visualizations correctly', () => {
```

Comment:
It might make sense to add a couple more tests. One with multiple sources to make sure the collection level associations are associated with the right source and one with multiple variables with one having visualizations to make sure the right variable gets the visualizations and the other does not.

Reply:
Oh, wait, no. No need to ingest for this test.